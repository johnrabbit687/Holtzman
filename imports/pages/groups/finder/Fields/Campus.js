import { Component, PropTypes } from "react";
import Forms from "../../../../components/@primitives/UI/forms";

const focusedInput = {
  border: "1px solid #f0f0f0",
  borderRadius: 7,
  boxShadow: "0px 2px 9px #DDD",
  backgroundColor: "#FFFFFF",
  display: "block",
  textTransform: "captialize",
  position: "relative",
  zIndex: 99,
};

const hiddenInput = {
  border: 0,
};

export default class Campus extends Component {
  state: {
    focused: boolean,
    campus: string,
    onload: boolean
  };

  static propTypes = {
    campuses: PropTypes.array.isRequired,
    selectedCampus: PropTypes.string.isRequired,
    campusOnChange: PropTypes.func.isRequired,
  };

  constructor(props: Object) {
    super(props);
    this.state = {
      focused: false,
      campus: "",
      onload: true,
    };

    this.setWrapperRef = this.setWrapperRef.bind(this);
    this.handleClickOutside = this.handleClickOutside.bind(this);
    // this.onClick = this.onClick.bind(this);
  }

  componentDidMount() {
    document.addEventListener("mousedown", this.handleClickOutside);
    document.addEventListener("touchstart", this.handleClickOutside);
  }

  componentWillReceiveProps(nextProps: Object) {
    if (nextProps.selectedCampus) {
      this.setState({
        campus: nextProps.selectedCampus,
      });
    }
  }

  componentWillUnmount() {
    document.removeEventListener("mousedown", this.handleClickOutside);
    document.removeEventListener("touchend", this.handleClickOutside);
  }

  setWrapperRef(node) {
    this.wrapperRef = node;
  }

  /**
   * Alert if clicked on outside of element
   */
  handleClickOutside(event) {
    if (this.wrapperRef && !this.wrapperRef.contains(event.target)) {
      this.setState({
        focused: false,
      });
    }
  }

  setFocus = () => {
    const focused = this.state.onload && this.props.selectedCampus ? false : !this.state.focused;

    this.setState({
      focused,
      onload: false,
    });
  };

  buttonToggle = () => {
    this.setState({
      focused: !this.state.focused,
    });
  };

  onClick = (e: Event) => {
    const campus = e.target.name;
    this.setState({
      campus,
    });

    this.props.campusOnChange(campus);
  };

  render() {
    const { campuses } = this.props;
    const { campus } = this.state;

    return (
      <div
        style={this.state.focused ? focusedInput : hiddenInput}
        className={"soft-double-top text-left soft-half-sides"}
        ref={this.setWrapperRef}
      >
        <Forms.Input
          classes={this.state.focused ? "soft-bottom" : ""}
          inputClasses={"outlined--dotted outlined--light h6 flush-bottom text-black"}
          style={{ textTransform: "capitalize" }}
          type="text"
          label={"Campus"}
          labelStyles={{ pointerEvents: "none" }}
          name="campus"
          defaultValue={campus}
          readOnly={this.state.focused ? "readonly" : ""}
          onFocus={this.setFocus}
          iconName={this.state.focused ? "arrowUp" : "arrowDown"}
          iconFill={this.state.focused ? "#6BAC43" : "#505050"}
          iconWidth={"24px"}
          iconHeight={"24px"}
          iconTitle={this.state.focused ? "Arrow Up Icon" : "Arrow Down Icon"}
          iconButtonToggle={this.buttonToggle}
        />
        <div
          className={`push-half-sides push-half-bottom ${!this.state.focused
            ? "visuallyhidden"
            : "display-inline-block"}`}
        >
          {/* weird SSR stuff here to investigate */}
          {campuses.map((c, i) =>
            <Forms.Checkbox
              classes={[
                "soft-half-bottom",
                "float-left@lap-and-up",
                "one-half@lap-and-up",
                "display-inline-block@lap-and-up",
              ]}
              defaultChecked={campus === c ? "defaultChecked" : ""}
              name={c}
              key={i}
              clicked={this.onClick}
            >
              <span className="soft-half-top" style={{ textTransform: "capitalize" }}>
                {c}
              </span>
            </Forms.Checkbox>
          )}
        </div>
      </div>
    );
  }
}
