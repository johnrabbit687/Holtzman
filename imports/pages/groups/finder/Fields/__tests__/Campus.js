import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Campus from "../Campus";

const defaultProps = {
  campuses: ["Greenville", "Anderson", "Aiken", "Northeast Columbia"],
  selectedCampus: "Anderson",
  onChange: jest.fn(),
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <Campus {...newProps} />;
};

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders with no campus", () => {
  const wrapper = shallow(
    generateComponent({
      selectedCampus: "",
    }),
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("on inital load don't show the modal with selectedCampus", () => {
  const wrapper = shallow(generateComponent());
  wrapper.instance().onFocus();
  expect(wrapper.state().focused).toBe(false);
  expect(wrapper.state().onload).toBe(false);
});

it("shows the modal with onFocus", () => {
  const wrapper = shallow(generateComponent());
  // onload prevents the modal from showing on inital load
  wrapper.setState({ onload: false });
  wrapper.instance().onFocus();
  expect(wrapper.state().focused).toBe(true);
});

it("shows the modal with onBlur", () => {
  const wrapper = shallow(generateComponent());
  wrapper.instance().onBlur();
  expect(wrapper.state().focused).toBe(false);
});

it("sets campus state when campus is clicked", () => {
  const wrapper = shallow(generateComponent());
  wrapper.instance().onClick({ target: { name: "aiken" } });
  expect(wrapper.state().campus).toEqual("aiken");
});
