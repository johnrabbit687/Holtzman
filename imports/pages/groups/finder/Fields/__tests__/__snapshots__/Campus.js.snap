exports[`test renders with no campus 1`] = `
<div
  className="soft-double-top text-left soft-half-sides"
  style={
    Object {
      "border": 0,
    }
  }>
  <Input
    autoCapitalize={false}
    autoComplete={false}
    autoCorrect={false}
    classes=""
    defaultValue=""
    ignoreLastPass={true}
    inputClasses="outlined--dotted outlined--light h6 flush-bottom text-black"
    label="Campus"
    labelStyles={
      Object {
        "pointerEvents": "none",
      }
    }
    name="campus"
    onBlur={[Function]}
    onFocus={[Function]}
    spellCheck={false}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
    type="text"
    validation={[Function]}>
    <a
      id="campusButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "paddingLeft": "5px",
          "position": "absolute",
          "right": "0",
          "top": "-5px",
        }
      }>
      <Svg
        fill="#505050"
        height="24px"
        name="arrowDown"
        title="Arrow Down Icon"
        width="24px" />
    </a>
  </Input>
  <div
    className="push-half-sides push-half-bottom visuallyhidden">
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Greenville">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Greenville
      </span>
    </Checkbox>
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Anderson">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Anderson
      </span>
    </Checkbox>
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Aiken">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Aiken
      </span>
    </Checkbox>
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Northeast Columbia">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Northeast Columbia
      </span>
    </Checkbox>
  </div>
</div>
`;

exports[`test renders with props 1`] = `
<div
  className="soft-double-top text-left soft-half-sides"
  style={
    Object {
      "border": 0,
    }
  }>
  <Input
    autoCapitalize={false}
    autoComplete={false}
    autoCorrect={false}
    classes=""
    defaultValue="Anderson"
    ignoreLastPass={true}
    inputClasses="outlined--dotted outlined--light h6 flush-bottom text-black"
    label="Campus"
    labelStyles={
      Object {
        "pointerEvents": "none",
      }
    }
    name="campus"
    onBlur={[Function]}
    onFocus={[Function]}
    spellCheck={false}
    style={
      Object {
        "textTransform": "capitalize",
      }
    }
    type="text"
    validation={[Function]}>
    <a
      id="campusButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "paddingLeft": "5px",
          "position": "absolute",
          "right": "0",
          "top": "-5px",
        }
      }>
      <Svg
        fill="#505050"
        height="24px"
        name="arrowDown"
        title="Arrow Down Icon"
        width="24px" />
    </a>
  </Input>
  <div
    className="push-half-sides push-half-bottom visuallyhidden">
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Greenville">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Greenville
      </span>
    </Checkbox>
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked="defaultChecked"
      name="Anderson">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Anderson
      </span>
    </Checkbox>
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Aiken">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Aiken
      </span>
    </Checkbox>
    <Checkbox
      classes={
        Array [
          "soft-half-bottom",
          "float-left@lap-and-up",
          "one-half@lap-and-up",
          "display-inline-block@lap-and-up",
        ]
      }
      clicked={[Function]}
      defaultChecked=""
      name="Northeast Columbia">
      <span
        className="soft-half-top"
        style={
          Object {
            "textTransform": "capitalize",
          }
        }>
        Northeast Columbia
      </span>
    </Checkbox>
  </div>
</div>
`;
